projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-send-email-preview.php

```diff
@@ -0,0 +1,124 @@
+		$this->version                         = 'v2';
+		$this->namespace                       = $this->base_api_path . '/' . $this->version;
+		$this->rest_base                       = '/send-email-preview';
+		$this->wpcom_is_wpcom_only_endpoint    = true;
+		$this->wpcom_is_site_specific_endpoint = true;
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Registers the routes for blogging prompts.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		$options = array(
+			'show_in_index'       => true,
+			'methods'             => 'POST',
+			// if this is not a wpcom site, we need to proxy the request to wpcom
+			'callback'            => ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? array( $this, 'send_email_preview' ) : array( $this, 'proxy_request_to_wpcom' ),
+			'permission_callback' => array( $this, 'permissions_check' ),
+			'args'                => array(
+				'id' => array(
+					'description' => __( 'Unique identifier for the post.', 'jetpack' ),
+					'type'        => 'integer',
+				),
+			),
+		);
+
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			$options
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to edit the post for a site.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return true|WP_Error True if the request has edit access, WP_Error object otherwise.
+	 */
+	public function permissions_check( $request ) {
+		if ( ! ( new Manager() )->is_user_connected() ) {
+			return new WP_Error(
+				'rest_cannot_send_email_preview',
+				__( 'Sorry, connect your user to be able to send email previews.', 'jetpack' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$request['context'] = 'edit';
+
+		return $this->get_item_permissions_check( $request );
+	}
+
+	/**
+	 * Sends an email preview of a post to the current user.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function send_email_preview( $request ) {
+		$post_id = $request['id'];
+		$post    = $this->get_post( $post_id );
+
+		// Return error if the post cannot be retrieved
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		// Check if the user's email is verified
+		if ( Email_Verification::is_email_unverified() ) {
+			return new WP_Error( 'unverified', __( 'Your email address must be verified.', 'jetpack' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		$current_user = wp_get_current_user();
+		$email        = $current_user->user_email;
+
+		// Try to create a new subscriber with the user's email
+		$subscriber = Blog_Subscriber::create( $email );
+		if ( ! $subscriber ) {
+			return new WP_Error( 'unverified', __( 'Could not create subscriber.', 'jetpack' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
The error description is vague. Does it mean user's email isn't verified?

Reply:
Edit: only now saw `Email_Verification::is_email_unverified()` check above this, so I'll need to better understand what this subscriber stuff does. :-)

Reply:
Aight 👍  yeah good to keep consistent. 

What happens with the `Blog_Subscriber::create` - what are we checking for with it?

Reply:
again this is copied, but seems like the only way to use `Subscription_Mailer` is to pass a $subscriber object 🤷‍♂️